test/rotoscope_test.rb

```diff
@@ -424,13 +424,22 @@ def test_dynamic_class_creation
     ], parse_and_normalize(contents)
   end
 
-  def test_dynamic_methods_in_blacklist
-    skip <<-FAILING_TEST_CASE
-      Return events for dynamically created methods (define_method, define_singleton_method)
-      do not have the correct stack frame information (the call of a dynamically defined method
-      is correctly treated as a Ruby :call, but its return must be treated as a :c_return)
-    FAILING_TEST_CASE
+  def test_block_defined_methods
+    contents = rotoscope_trace { Example.apply("my value!") }
 
+    assert_equal [
+      { event: "call", entity: "Example", method_name: "apply", method_level: "class", filepath: "/rotoscope_test.rb", lineno: -1 },
+      { event: "call", entity: "Example", method_name: "monad", method_level: "class", filepath: "/rotoscope_test.rb", lineno: -1 },
+      { event: "call", entity: "Example", method_name: "contents", method_level: "class", filepath: "/monadify.rb", lineno: -1 },
+      { event: "return", entity: "Example", method_name: "contents", method_level: "class", filepath: "/monadify.rb", lineno: -1 },
+      { event: "call", entity: "Example", method_name: "contents=", method_level: "class", filepath: "/monadify.rb", lineno: -1 },
+      { event: "return", entity: "Example", method_name: "contents=", method_level: "class", filepath: "/monadify.rb", lineno: -1 },
+      { event: "return", entity: "Example", method_name: "monad", method_level: "class", filepath: "/rotoscope_test.rb", lineno: -1 },
+      { event: "return", entity: "Example", method_name: "apply", method_level: "class", filepath: "/rotoscope_test.rb", lineno: -1 },
+    ], parse_and_normalize(contents)
+  end
```

Comment:
Added this test after @dylanahsmith's comments. Had to change a lot of the logic around in `log_rotoscope_trace`, but this test now passes 🎉